lib/ansible/modules/cloud/amazon/ec2_placement_group_facts.py

```diff
@@ -0,0 +1,150 @@
+        will be no corresponding entry in the result; no error will be raised.
+    required: false
+    default: []
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details or the AWS region,
+# see the AWS Guide for details.
+
+# List all placement groups.
+- ec2_placement_group_facts:
+  register: all_ec2_placement_groups
+
+# List two placement groups.
+- ec2_placement_group_facts:
+  names:
+     - my-cluster
+     - my-other-cluster
+  register: specific_ec2_placement_groups
+
+
+'''
+
+
+RETURN = '''
+placement_groups:
+  description: Placement group attributes
+  returned: always
+  type: complex
+  contains:
+    name:
+      description: PG name
+      type: string
+      sample: my-cluster
+    state:
+      description: PG state
+      type: string
+      sample: "available"
+    strategy:
+      description: PG strategy
+      type: string
+      sample: "cluster"
+
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (AnsibleAWSError,
+                                      connect_to_aws,
+                                      boto3_conn,
+                                      ec2_argument_spec,
+                                      get_aws_connection_info,
+                                      get_ec2_security_group_ids_from_names)
+
+
+def get_placement_groups_details(connection, module):
+    names = module.params.get("names")
+    if len(names) > 0:
+        response = connection.describe_placement_groups(
+            Filters=[{
+                "Name": "group-name",
+                "Values": names
+            }])
+    else:
+        response = connection.describe_placement_groups()
+
+    results = []
+    for placement_group in response['PlacementGroups']:
+        results.append({
+            "name": placement_group['GroupName'],
+            "state": placement_group['State'],
+            "strategy": placement_group['Strategy'],
+        })
+    return results
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            names=dict(type='list', default=[])
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(
+        module, boto3=True)
+
+    if region:
+        try:
+            connection = boto3.client('ec2', region, **aws_connect_params)
+        except (boto3.exceptions.NoAuthHandlerFound, AnsibleAWSError) as e:
+            module.fail_json(msg=str(e))
+    else:
+        module.fail_json(msg="region must be specified")
```

Comment:
boto3_conn now copes with RegionNotFound errors for you

Reply:
No worries - addressed (hopefully all) these in this module and in ec2_placement_groups.py (separate PR).